sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,773 @@
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_program32(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint32_t address, uint32_t value);
+
+/**
+ * Schedules a 64-bit write on the Direct Access Interface.
+ *
+ * Writes are performed at the partition granularity; `address` should be given
+ * relative to the start of `partition`. An error is returned for out-of-bounds
+ * access.
+ *
+ * Furthermore, `address` must be eight-byte-aligned, and `partition` must be
+ * a secret partition. An error is returned if neither condition is met.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to program.
+ * @param address A partition-releative address to program.
+ * @param value The value to program into the OTP.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_program64(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint32_t address, uint64_t value);
+
+/**
+ * Schedules a hardware digest operation on the Direct Access Interface.
+ *
+ * This operation will also lock writes for the given partition.
+ *
+ * `partition` must not be an SW partition; those partitions do not have
+ * hardware-managed digests. An error is returned for SW partitions and the
+ * lifecycle state partition.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to digest and lock.
+ * @return The result of the operation.
+ */
+// Open Q: do we want a SW version, as a convenience for programming the digest
+// of a SW partition?
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_hw_digest(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition);
+
+/**
+ * Gets the buffered digest value for the given partition.
+ *
+ * Note that this value is only updated when the device is reset. A value of
+ * zero indicates that the digest has not been burned in yet (and, therefore,
+ * that the partition is writeable).
+ *
+ * The lifecycle partition does not have a digest and will result in an error
+ * being returned.
+ *
+ * @param otp An OTP handle.
+ * @param partition The parittion to get a digest for.
+ * @param[out] digest Out-param for the digest.
+ * @return The result of the operation.
+ */
+// Open Q: should a value of zero be returned as an error, indicating that the
+// digest doesn't exist yet?
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_result_t dif_otp_ctrl_get_digest(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint64_t *digest);
+
+/**
+ * Gets the buffered lifecycle state.
+ *
+ * TODO(mcyoung): More docs when we decide on the interface for this function.
+ */
+// Open Q: do we want to wrap `state` in a struct? Right now it just gets
+// decayed into a pointer so the length parameter is mostly a lie.
+//
+// Alternatively, should this return a bit vector indicating which lifecycle
+// state half-words have been burned?
```

Comment:
I honestly feel more partial to the bit vector, seeing as the LC state is just that, with each half-word being used to encode whether a bit has been written. It would be a more compact interface, at any rate. WDYT?

Reply:
For the moment, just return the state array in a struct. Once we have a better idea how this interacts with lifecycle management, we can revisit this one function's API.

Reply:
Done.